packages/flutter_tools/test/integration.shard/test_data/hot_reload_websocket_test_common.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/web/web_device.dart' show WebServerDevice;
+
+import '../../src/common.dart';
+import '../test_driver.dart';
+import '../test_utils.dart';
+import 'hot_reload_project.dart';
+
+void testAll({List<String> additionalCommandArgs = const <String>[]}) {
+  group('WebSocket DWDS connection'
+      '${additionalCommandArgs.isEmpty ? '' : ' with args: $additionalCommandArgs'}', () {
+    // Test configuration constants
+    const Duration debugUrlTimeout = Duration(seconds: 20);
+    const Duration appStartTimeout = Duration(seconds: 15);
+    const Duration hotReloadTimeout = Duration(seconds: 10);
+
+    // Chrome executable paths (macOS primary, fallback)
+    const List<String> chromePaths = [
+      '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome',
```

Comment:
I'd use `findChromeExecutable` from `chrome.dart` instead of rolling your own here. Could `ChromiumLauncher` be repurposed here as well instead of starting a process separately?